cli/azd/cmd/provision.go

```diff
@@ -129,34 +130,32 @@ func (p *provisionAction) Run(ctx context.Context) (*actions.ActionResult, error
 		defaultTitleNote = "This is a preview. No changes will be applied to your Azure resources."
 	}
 
+	p.console.MessageUxItem(ctx, &ux.MessageTitle{
+		Title:     defaultTitle,
+		TitleNote: defaultTitleNote},
+	)
+
 	// Get Subscription to Display in Command Title Note
 	subscriptions, subErr := p.subManager.GetSubscriptions(ctx)
 	if subErr == nil {
 		// Find subscription name
 		for _, sub := range subscriptions {
 			if sub.Id == p.env.GetSubscriptionId() {
-				messageFormat := "Provisioning Azure resources in subscription (%s) %s and location (%s) can take some time"
-				if previewMode {
-					messageFormat = "This is a preview. No changes will be applied to your Azure resources in subscription (%s) %s " +
-						"and location (%s)."
+				location, err := p.subManager.GetLocation(ctx, p.env.GetSubscriptionId(), p.env.GetLocation())
+				if err != nil {
+					log.Printf("failed getting location: %v", err)
 				}
-				// Formate the note
-				defaultTitleNote = fmt.Sprintf(
-					messageFormat,
-					sub.Name,
-					sub.Id,
-					p.env.GetLocation(),
+				p.console.MessageUxItem(ctx, &ux.EnvironmentDetails{
+					Subscription: fmt.Sprintf("%s (%s)", sub.Name, sub.Id),
+					Location:     location.DisplayName},
 				)
 				break
```

Comment:
I'm not quite sure this will work for all provision providers.  Specifically in ADE there isn't any known subscription or location till after provisioning is complete.  It feels like any additional messaging should be left to the provision provider and not the command.

Reply:
if p.subManager.GetSubscriptions(ctx) returns error, or empty list, azd won't add any message.
If p.env.GetSubscriptionId() returns empty, that's the same.

So, the message will only be there when it is possible to be displayed (subscriptions available and a subscription id on the .env).  If that's never the case for ADE, it will be never displayed.
If that's only true for ADE after provisioning, then the message will be there only after provisioning